src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenAResolveTargetingPackAssetsTask.cs

```diff
@@ -0,0 +1,424 @@
+                .Should().BeEquivalentTo(new[]
+                {
+                    "RuntimeFramework1",
+                    "RuntimeFramework2",
+                });
+        }
+
+            [Fact]
+        public void Given_Passing_ResolvedTargetingPacks_It_Passes_Again_With_Cached_Results()
+        {
+            ResolveTargetingPackAssets task1 = InitializeTask(out string packageDirectory);
+
+            // Save off that build engine to inspect and reuse
+            MockBuildEngine4 buildEngine = (MockBuildEngine4)task1.BuildEngine;
+
+            task1.Execute().Should().BeTrue();
+
+            buildEngine.RegisteredTaskObjectsQueries.Should().Be(2,
+                because: "there should be a lookup for the overall and the specific targeting pack");
+
+            buildEngine.RegisteredTaskObjects.Count.Should().Be(2,
+                because: "there should be a cache entry for the overall lookup and for the specific targeting pack");
+
+            ResolveTargetingPackAssets task2 = InitializeTask(packageDirectory, buildEngine);
+
+            task2.Execute().Should().BeTrue();
+
+            buildEngine.RegisteredTaskObjectsQueries.Should().Be(3,
+                because: "there should be a hit on the overall lookup this time through");
+
+            buildEngine.RegisteredTaskObjects.Count.Should().Be(2,
+                because: "the cache keys should match");
+        }
+
+        [Fact]
+        public void Given_Passing_ResolvedTargetingPacks_A_Different_Language_Parses_Again()
+        {
+            ResolveTargetingPackAssets task1 = InitializeTask(out string packageDirectory);
+
+            // Save off that build engine to inspect and reuse
+            MockBuildEngine4 buildEngine = (MockBuildEngine4)task1.BuildEngine;
+
+            task1.Execute().Should().BeTrue();
+
+            buildEngine.RegisteredTaskObjectsQueries.Should().Be(2,
+                because: "there should be a lookup for the overall and the specific targeting pack");
+
+            buildEngine.RegisteredTaskObjects.Count.Should().Be(2,
+                because: "there should be a cache entry for the overall lookup and for the specific targeting pack");
+
+            ResolveTargetingPackAssets task2 = InitializeTask(packageDirectory, buildEngine);
+
+            task2.ProjectLanguage = "F#";
+
+            task2.Execute().Should().BeTrue();
+
+            buildEngine.RegisteredTaskObjectsQueries.Should().Be(4,
+                because: "there should be no hits on the overall or targeting pack lookup this time through");
+
+            buildEngine.RegisteredTaskObjects.Count.Should().Be(4,
+                because: "there should be distinct results for C# and F#");
+        }
+
+        private ResolveTargetingPackAssets InitializeTask(out string mockPackageDirectory)
+        {
+            mockPackageDirectory = Path.Combine(Path.GetTempPath(), "dotnetSdkTests", Path.GetRandomFileName());
```

Comment:
I'm not a fan of putting test state in temporary directories that don't get cleaned up and are hard to find for debugging.  Can we make this test class derive from `SdkTest` and then use `_testAssetsManager.CreateTestDirectory()` or something?

Most of the task unit tests don't derive from `SdkTest`, but `GivenAResolveRuntimePackAssetsTask` does, so it should be OK.

Reply:
This was preexisting but sure.

Reply:
Is 289ee935ca what you were thinking of?

Reply:
Pretty much, but I think you'll need to update it to use a different directory each time it's called as part of the same test.  We generally do this with an `identifier` parameter that you can pass through to CreateTestDirectory.

When running in CI, we will fail if it tries to create a test directory that already exists.  This helps prevent flakiness where different tests end up using the same test directory.  We don't do that locally, since otherwise every time you re-ran a test it would fail.

Reply:
Done in 5f64f82fa9 but right now no test calls it twice so I didn't actually call it. Do you want me to pass it in in every call anyway?

Reply:
OK, I got confused by the different overloads of `InitializeTask`: https://github.com/dotnet/sdk/blob/5f64f82fa9faf86654b837be4c0b695e64f4caf2/src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenAResolveTargetingPackAssetsTask.cs#L74-L87

If no test actually calls it twice then you don't even need to take the identifier as an argument, I just thought the tests were re-creating the same directory and that was going to cause failures in CI.

Reply:
Pushed a revert + some strategic renames to hopefully clarify for the future.

Reply:
Thanks.  I already signed off so feel free to merge.